include/exadg/incompressible_navier_stokes/user_interface/viscosity_model_data.h

```diff
@@ -0,0 +1,131 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_INCOMPRESSIBLE_NAVIER_STOKES_USER_INTERFACE_VISCOSITY_MODEL_DATA_H_
+#define INCLUDE_EXADG_INCOMPRESSIBLE_NAVIER_STOKES_USER_INTERFACE_VISCOSITY_MODEL_DATA_H_
+
+namespace ExaDG
+{
+namespace IncNS
+{
+/*
+ *  Turbulence model data.
+ */
+struct TurbulenceModelData
+{
+  TurbulenceModelData()
+    : turbulence_model(TurbulenceEddyViscosityModel::Undefined), is_active(false), constant(0.0)
+  {
+  }
+
+  TurbulenceEddyViscosityModel turbulence_model;
+  bool                         is_active;
+  double                       constant; // model constant
+
+  void
+  check() const
+  {
+    AssertThrow(is_active, dealii::ExcMessage("Turbulence model is inactive."));
+    AssertThrow(constant > 1e-20, dealii::ExcMessage("Parameter must be greater than zero."));
+  }
+};
+
+/*
+ * Generalized Newtonian model data.
+ */
+struct GeneralizedNewtonianModelData
+{
+  GeneralizedNewtonianModelData()
+    : generalized_newtonian_model(GeneralizedNewtonianViscosityModel::Undefined),
+      is_active(false),
+      viscosity_margin(0.0),
+      kappa(0.0),
+      lambda(0.0),
+      a(0.0),
+      n(0.0)
+  {
+  }
+
+  GeneralizedNewtonianViscosityModel generalized_newtonian_model;
+  bool                               is_active;
+
+  // parameters of generalized Carreau-Yasuda model
+  double viscosity_margin;
+  double kappa;
+  double lambda;
+  double a;
+  double n;
```

Comment:
One thing I find really cool is the in-class initialization, where you don't have to write the constructor if it is boring like this one. Instead you can write the following. Totally up to you to do it though.
```suggestion
struct GeneralizedNewtonianModelData
{
  GeneralizedNewtonianViscosityModel generalized_newtonian_model{GeneralizedNewtonianViscosityModel::Undefined};
  bool                               is_active{false};

  // parameters of generalized Carreau-Yasuda model
  double viscosity_margin{0.0};
  double kappa{0.0};
  double lambda{0.0};
  double a{0.0};
  double n{0.0};
```

Reply:
Since the current version is the current "ExADG-style", I am ok with the current version. But it's a good hint @btemuer.

Reply:
Do we allow both in the future?
The second is a bit shorter, so  I would always do something like this in the future if we allow that?